src/cdk/overlay/position/flexible-connected-position-strategy.ts

```diff
@@ -491,13 +501,26 @@ export class FlexibleConnectedPositionStrategy implements PositionStrategy {
       x = pos.originX == 'start' ? startX : endX;
     }
 
+    // When zooming in Safari the container rectangle contains negative values for the position
+    // and we need to re-add them to the calculated coordinates.
+    if (containerRect.left < 0) {
+      x -= containerRect.left;
+    }
+
     let y: number;
     if (pos.originY == 'center') {
       y = originRect.top + originRect.height / 2;
     } else {
       y = pos.originY == 'top' ? originRect.top : originRect.bottom;
     }
 
+    // Normally the containerRect's top value would be zero, however when the overlay is attached to an input
+    // (e.g. in an autocomplete), mobile browsers will shift everything in order to put the input in the middle
+    // of the screen and to make space for the virtual keyboard. We need to account for this offset,
+    // otherwise our positioning will be thrown off.
+    // Additionally, when zooming in Safari this fixes the vertical position.
+    y -= containerRect.top;
```

Comment:
Just to be on the safe side, shouldn't we check whether `top` is negative like we do above with `left`?

Reply:
I added the check. The reason I did not add the check was that there was no check before when the expression was in the method `_getExactOverlayY` (there to fix virtual keyboard offset).